Whiskey/Tasks/Pester5.ps1

```diff
@@ -0,0 +1,195 @@
+
+            foreach( $pathItem in $path )
+            {
+                if( $Exclude )
+                {
+                    $excluded = $false
+                    foreach( $exclusion in $Exclude )
+                    {
+                        if( $pathItem -like $exclusion )
+                        {
+                            Write-WhiskeyVerbose -Context $TaskContext -Message ('EXCLUDE  {0} -like    {1}' -f $pathItem,$exclusion)
+                            $excluded = $true
+                        }
+                        else
+                        {
+                            Write-WhiskeyVerbose -Context $TaskContext -Message ('         {0} -notlike {1}' -f $pathItem,$exclusion)
+                        }
+                    }
+
+                    if( $excluded )
+                    {
+                        continue
+                    }
+                }
+
+                if( $scriptItem -is [String] )
+                {
+                    Write-Output $pathItem
+                    continue
+                }
+
+                if( $scriptItem | Get-Member -Name 'Keys' )
+                {
+                    $newScriptItem = $scriptItem.Clone()
+                    $newScriptItem['Path'] = $pathItem
+                    Write-Output $newScriptItem
+                }
+            }
+        }
+    }
+
+    if( -not $PesterConfiguration.Run.Container.Path )
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message ('Found no tests to run.')
+        return
+    }
+
+    $pesterManifestPath = $PesterModuleInfo.Path
+
+    $cmdArgList = @(
+        (Get-Location).Path,
+        $pesterManifestPath,
+        $PesterConfiguration,
+        @{
+            'VerbosePreference' = $VerbosePreference;
+            'DebugPreference' = $DebugPreference;
+            'ProgressPreference' = $ProgressPreference;
+            'WarningPreference' = $WarningPreference;
+            'ErrorActionPreference' = $ErrorActionPreference;
+        }
+    )
+
+    $cmdName = 'Invoke-Command'
+    if( $AsJob )
+    {
+        $cmdName = 'Start-Job'
+    }
+
+    $result = & $cmdName -ArgumentList $cmdArgList -ScriptBlock {
+        param(
+            [String] $WorkingDirectory,
+
+            [String] $PesterManifestPath,
+
+            [hashtable] $PesterConfiguration,
+
+            [hashtable] $Preference
+        )
+        
+        Set-Location -Path $WorkingDirectory
+
+        $VerbosePreference = 'SilentlyContinue'
+        Import-Module -Name $PesterManifestPath -Verbose:$false -WarningAction Ignore
+
+        $VerbosePreference = $Preference['VerbosePreference']
+        $DebugPreference = $Preference['DebugPreference']
+        $ProgressPreference = $Preference['ProgressPreference']
+        $WarningPreference = $Preference['WarningPreference']
+        $ErrorActionPreference = $Preference['ErrorActionPreference']
+
+        # New Pester5 Configuration
+        $configuration = [PesterConfiguration]@{

```

Comment:
Because of the ArrayList problem, you'll need to write a function that iterates through the hashtable recursively and change any values that are arraylists to normal arrays.